http-api/src/main/java/bisq/http_api/ApiTorOnionService.java

```diff
@@ -37,23 +39,28 @@ public class ApiTorOnionService implements Service {
     private final Path appDataDirPath;
     private final KeyBundleService keyBundleService;
     private final String identityTag;
+    @Getter
+    private final boolean publishOnionService;
+    @Getter
+    private Optional<Address> publishedAddress = Optional.empty();
 
     public ApiTorOnionService(Path appDataDirPath,
                               SecurityService securityService,
                               NetworkService networkService,
                               int port,
-                              String identityTag) {
+                              String identityTag,
+                              boolean publishOnionService) {
         this.appDataDirPath = appDataDirPath;
         keyBundleService = securityService.getKeyBundleService();
         this.identityTag = identityTag;
         this.networkService = networkService;
         this.port = port;
+        this.publishOnionService = publishOnionService;
     }
 
     @Override
     public CompletableFuture<Boolean> initialize() {
-        if (networkService.findServiceNode(TransportType.TOR).isEmpty()) {
-            // If tor is not used we don't do anything
+        if (!publishOnionService) {
```

Comment:
The `if (networkService.findServiceNode(TransportType.TOR).isEmpty()) {` is still useful IMO.